compiler/stmt.py

```diff
@@ -41,6 +65,40 @@ def __init__(self, block_):
     self.block = block_
     self.writer = util.Writer()
     self.expr_visitor = expr_visitor.ExprVisitor(self.block, self.writer)
+    self.parser_flags = 0
+    self.did_future_pass = False
+    self.future_lineno = 0
+
+  def visit(self, node):
+    root = node
+    # If this is the module node, do an initial pass through the module body's
```

Comment:
Sorry to bring this up late, but what do you think about having a separate FutureVisitor (or something) class that performs this logic ahead of time. The future flags can then be computed and passed into the ModuleBlock constructor (somewhere around [here](https://github.com/google/grumpy/blob/master/tools/grumpc#L53)) and accessed by the StatementVisitor.

The nice thing about that is it reduces the amount of state managed by the StatementVisitor. I'm concerned about increasing the complexity of StatementVisitor because it could easily become a catch-all and get out of control given its centrality.